File: 18228.py
N = int(input())
ice = list(map(int, input().split()))

index = 0
min_left = 1000000001
while ice[index] != -1:
    if ice[index] < min_left:
        min_left = ice[index]
    index += 1

min_right = 1000000001
index += 1
while index < N:
    if ice[index] < min_right:
        min_right = ice[index]
    index += 1

print(min_left + min_right)

File: 18228.cpp
#include <iostream>
using namespace std;

int ice[200000];

int main() {
	int N; cin >> N;
	for (int i = 0; i < N; i++)
		cin >> ice[i];

	int index = 0;
	int min_left = 1000000001;
	for (; ice[index] != -1; index++)
		if (ice[index] < min_left)
			min_left = ice[index];

	int min_right = 1000000001;
	for (index++; index < N; index++)
		if (ice[index] < min_right)
			min_right = ice[index];

	cout << min_left + min_right << endl;
}
